pkg/network/nr/net_resource.go

```diff
@@ -249,19 +249,21 @@ func (nr *NetResource) Delete() error {
 	return nil
 }
 
-func (nr *NetResource) routes() ([]*netlink.Route, error) {
-	routes := make([]*netlink.Route, 0, len(nr.resource.Peers)+1)
+func (nr *NetResource) routes() ([]netlink.Route, error) {
```

Comment:
I do not understand this change. The commit message seems to only talk about pointer. 
But there is actually a logic change here. 
The gateway of the route is now different. Is this change valid ?

Reply:
yes there were 2 changes in this code, the first fixed a bug related to how golang handles loops over slices and pointers, the second changed some of the logic to properly set up all routes as provided in the reservation and remove the catch all rule

Reply:
:+1: